controller.js

```diff
@@ -1,12 +1,27 @@
-const { selectTopics } = require("./model")
+const { selectTopics, selectArticleById } = require("./model")
 
 
 exports.getTopics = (req, res, next) => {
-    selectTopics().then((result) =>{
-        res.status(200).send({result})
+    selectTopics().then((topics) =>{
+        res.status(200).send({topics})
     })
     .catch((err)=>{
         next(err)
     })
-    
+}
+
+exports.getArticleById = (req, res ,next) =>{
+    const {article_id} = req.params
+    if(isNaN(article_id)){
```

Comment:
You might not need this since PSQL will through its own error if the article_id is not a number, you should deal with this in error handling. Also if you were to do checks like this, they would go in the model since the the only job of the controller is to pass data around.